libcxx/include/stdatomic.h

```diff
@@ -121,7 +121,7 @@ using std::atomic_signal_fence                         // see below
 #  pragma GCC system_header
 #endif
 
-#if defined(__cplusplus) && _LIBCPP_STD_VER >= 23
+#if defined(__cplusplus)
```

Comment:
Since this header from libc++ is supposed to only be included in C++ mode, we should always take this branch and the `#else` should be unnecessary. Do you agree? The `#else` was previously useful because it handled the pre-C++23 behavior, but it wouldn't be useful anymore with this patch.

Reply:
Many of the libc++ C headers also have `defined(__cplusplus)`. (For example, `stdlib.h`.) It has mattered in the past for Android's Soong build system because IIRC it adds libc++ headers to the include path even when compiling a C source file.

Reply:
Nevermind, I was under the wrong impression that `stdatomic.h` was only provided in C++. It's also a C header, so of course it would need this `#if defined(__cplusplus)`.